ensime_shared/client.py

```diff
@@ -249,7 +249,7 @@ def send_at_position(self, what, useSelection, where="range"):
         self.log.debug('send_at_position: in')
         b, e = self.editor.selection_pos() if useSelection else self.editor.word_under_cursor_pos()
         self.log.debug('useSelection: {}, beg: {}, end: {}'.format(useSelection, b, e))
-        self.send_at_point_req(what, self.editor.path(), b[0], b[1], e[0], e[1], where)
+        self.send_at_range_req(what, self.editor.path(), b[0], b[1], e[0], e[1], where)
```

Comment:
I understand trying to keep changes minimal, but if this is now the only call site of the renamed `send_at_range_req`, could we drop its `where` parameter? I.e. branch in `send_at_position` if it was called with `where='range'` vs `'point'`. Does it sometimes need to be called with `where='point'` to send a message with `'point': {'from': …, 'to': …}`?

My concern with it is, previously you had one method `send_at_point_req` which could be called with either "point" or "range"… okay, that's a little bit weird to understand, but now you have two methods which *seem* to helpfully separate point vs. range yet actually you still sometimes might need to call the range one with a "point" argument! (Do you?)

I had started modeling protocol messages as `namedtuple`s in my separate client project but the protocol was changing more than I was willing to keep up with… if that's slowed down maybe we could start doing that in ensime-vim in the short term (not in this PR of course).